### PR TITLE
Registry Service DID

### LINKED ISSUE #44: Registration Service - Dataspace Authority DID
Feature agera-edc/MinimumViableDataspace#177
After agera-edc/MinimumViableDataspace#186 

## Description

Dataspace authority should have DID document related to it.

## Acceptance Criteria

- [x] Well-known DID endpoint for registration service deployed.

## Tasks

Same as acceptance criteria

### LINKED ISSUE #185: Registration Service - Dataspace Authority DID
Feature agera-edc/MinimumViableDataspace#177
After agera-edc/MinimumViableDataspace#186 

## Description

Dataspace authority should have DID document related to it.

## Acceptance Criteria

- [x] Well-known DID endpoint for registration service deployed.

## Tasks

Same as acceptance criteria

### PR DESCRIPTION
**Pull request by cpeeyush**
- Adds Well-known DID endpoint for registration service during deployment.

Closes agera-edc/MinimumViableDataspaceFork#44

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -184,6 +191,8 @@ jobs:
           echo "::set-output name=app_insights_connection_string::${app_insights_connection_string}"
           REGISTRY_HOST=$(terraform output -raw registry_host)
           echo "REGISTRY_HOST=$REGISTRY_HOST" >> $GITHUB_ENV
+          DID_HOST=$(terraform output -raw did_host)
+          echo "DID_HOST=$DID_HOST" >> $GITHUB_ENV
```
Comment (algattik) [inline .github/workflows/deploy.yaml:195]: nit: for consistency we could use `set-output` here and `{{ steps.runterraform.outputs.did_host }}` further down

File: deployment/terraform/dataspace/main.tf [hunk 2]
```diff
@@ -112,3 +114,86 @@ resource "azurerm_container_group" "registry-service" {
...
+    "@context" = ["https://www.w3.org/ns/did/v1",
+      {
+        "@base" = local.did_url
+      }
+    ],
+    "service" : [
+      {
+        "id" : "#identity-hub-url",
+        "type" : "IdentityHub",
+        "serviceEndpoint" : "http://dummy"
```
Comment (algattik) [inline deployment/terraform/dataspace/main.tf]: why add this block with dummy URL? the authority will not have an IH anyway, AFAIK

Comment (cpeeyush) [reply deployment/terraform/dataspace/main.tf]: As per the spec it has an identity hub. https://github.com/Metaform/mvd/blob/main/registration-service/registration-service-tech-spec.md

**Review by algattik — CHANGES_REQUESTED** [review]
comment inline

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (algattik) [inline .github/workflows/deploy.yaml]: nit: for consistency with https://github.com/agera-edc/MinimumViableDataspace/pull/210/files, consider using an output here instead (`set-output`)

File: deployment/terraform/dataspace/main.tf (hunk 2 above)
Comment (marcgs) [reply deployment/terraform/dataspace/main.tf]: AFAIK there are no explicit authorization requirements for the Registry Service so far, which would be the only reason to have an Identity Hub. To keep things simple for the MVD I would leave it out.

Comment (cpeeyush) [reply deployment/terraform/dataspace/main.tf]: okay as per the discussion I have removed it.

File: deployment/terraform/dataspace/main.tf [hunk 3]
```diff
@@ -112,3 +114,79 @@ resource "azurerm_container_group" "registry-service" {
...
+resource "azurerm_storage_blob" "did" {
+  name                 = ".well-known/did.json" # `.well-known` path is defined by did:web specification
+  storage_account_name = azurerm_storage_account.did.name
+  # Create did blob only if public_key_jwk_file is provided. Default public_key_jwk_file value is null.
+  count                  = var.public_key_jwk_file == null ? 0 : 1
+  storage_container_name = "$web" # container used to serve static files (see static_website property on storage account)
+  type                   = "Block"
+  source_content = jsonencode({
+    id = local.did_url
+    "@context" = ["https://www.w3.org/ns/did/v1",
```
Comment (algattik) [inline deployment/terraform/dataspace/main.tf]: nit

```suggestion
    "@context" = [
    "https://www.w3.org/ns/did/v1",
```

**Review by algattik — APPROVED** [review]

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (cpeeyush) [reply .github/workflows/deploy.yaml]: updated.

**Review by marcgs — APPROVED** [review]

**Pull request merged**